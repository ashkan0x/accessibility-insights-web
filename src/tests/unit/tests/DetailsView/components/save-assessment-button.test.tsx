// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { UserConfigMessageCreator } from 'common/message-creators/user-config-message-creator';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { AssessmentActionMessageCreator } from 'DetailsView/actions/assessment-action-message-creator';
import {
    SaveAssessmentButton,
    SaveAssessmentButtonProps,
} from 'DetailsView/components/save-assessment-button';
import * as React from 'react';
import {
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, It, Mock } from 'typemoq';

jest.mock('@fluentui/react-components', () => ({
    ...jest.requireActual('@fluentui/react-components'),
    makeStyles: () => () => ({}),
}));

jest.mock('common/components/controls/insights-command-button');

describe('SaveAssessmentButton', () => {
    mockReactComponents([InsightsCommandButton]);
    let propsStub: SaveAssessmentButtonProps;
    let assessmentActionMessageCreatorMock: IMock<AssessmentActionMessageCreator>;
    let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
    let userConfigurationStoreData: UserConfigurationStoreData;
    const handleSaveAssessmentButtonClickMock =
        Mock.ofType<(event: React.MouseEvent<any>) => void>();

    beforeEach(() => {
        assessmentActionMessageCreatorMock = Mock.ofType<AssessmentActionMessageCreator>();
        userConfigurationStoreData = {
            showSaveAssessmentDialog: true,
        } as UserConfigurationStoreData;
        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
        propsStub = {
            deps: {
                getAssessmentActionMessageCreator: () => assessmentActionMessageCreatorMock.object,
                userConfigMessageCreator: userConfigMessageCreatorMock.object,
            },
            download: 'download',
            href: 'url',
            userConfigurationStoreData,
            handleSaveAssessmentButtonClick: handleSaveAssessmentButtonClickMock.object,
        };
    });

    it('should render per snapshot', async () => {
        const wrapper = render(<SaveAssessmentButton {...propsStub} />);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });

    it('render Save assement button with props to match snapshot', () => {
        useOriginalReactElements('common/components/controls/insights-command-button', [
            'InsightsCommandButton',
        ]);
        const { asFragment } = render(<SaveAssessmentButton {...propsStub} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call save assessment button click method on click', async () => {
        useOriginalReactElements('common/components/controls/insights-command-button', [
            'InsightsCommandButton',
        ]);
        handleSaveAssessmentButtonClickMock.setup(m => m(It.isAny())).verifiable();
        const wrapper = render(<SaveAssessmentButton {...propsStub} />);
        await userEvent.click(wrapper.getByRole('button'));
        handleSaveAssessmentButtonClickMock.verifyAll();
    });
});
