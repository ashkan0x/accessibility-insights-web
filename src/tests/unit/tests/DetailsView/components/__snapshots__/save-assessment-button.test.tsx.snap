// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAssessmentButton should render per snapshot 1`] = `
<DocumentFragment>
  <mock-insightscommandbutton
    as="a"
    deps="[object Object]"
    download="download"
    href="url"
    insightscommandbuttoniconprops="[object Object]"
    userconfigurationstoredata="[object Object]"
  >
    Save assessment
  </mock-insightscommandbutton>
</DocumentFragment>
`;
